File: Maths_And_Stats/Number_Theory/Strong_Number/strong_number.c
#include <stdio.h>  
int main()  
{  
    int n;  
    int sum=0;  
    printf("Enter a number");  
    scanf("%d",&n);  
    int k=n;  
    int r;  
    while(k!=0)  
    {  
        r=k%10;  
        int f=fact(r);  
        k=k/10;  
        sum=sum+f;  
    }  
    if(sum==n)  
    {  
        printf("\nNumber is a strong");  
    }  
    else  
    {  
        printf("\nNumber is not a strong");  
    }  
    return 0;  
}  
int fact(int r)  
{  
    int mul=1;  
    for(int i=1;i<=r;i++)  
    {  
        mul=mul*i;  
    }  
    return mul;  
      
}  

File: Maths_And_Stats/Number_Theory/Strong_Number/strong_number.py
n= int(input("Enter the number : "))
sum = 0
temp = n
while (n):
    i = 1
    fact = 1
    rem = int(n % 10)

while(i <= rem):
    fact = fact * i
    i = i + 1
    sum = sum + fact
    n = int(n / 10)

if(sum == temp):
    print(temp,end = "")
    print(" is a strong number")
else:
    print(temp,end = "") 
    print(" is not a strong number")

